targets/arm/mikroe/sam/src/adc/implementations/implementation_1/hal_ll_adc.c

```diff
@@ -0,0 +1,546 @@
+static hal_ll_adc_handle_register_t hal_ll_module_state[ADC_MODULE_COUNT] = { (handle_t *) NULL, (handle_t *) NULL, false };
+
+/**
+ *  ADC modules
+ */
+typedef enum
+{
+    HAL_LL_MODULE_ADC_DEFAULT = 0xFF,
+    HAL_LL_MODULE_ADC_0 = 0,
+    HAL_LL_MODULE_ADC_1 = 1,
+
+    HAL_LL_MODULE_ADC_NONE
+} hal_ll_adc_module_number_t;
+
+/**
+ *  Return values
+ */
+typedef enum
+{
+    HAL_LL_ADC_SUCCESS = 0,
+    HAL_LL_ADC_WRONG_PIN,
+
+    HAL_LL_ADC_UNSUPPORTED_RESOLUTION = 1100,
+    HAL_LL_ADC_UNSUPPORTED_VREF,
+    HAL_LL_ADC_WRONG_CHANNEL,
+    HAL_LL_ADC_ERROR = (-1)
+} hal_ll_adc_err_t;
+
+/*!< @brief AFEC register structure. */
+typedef struct
+{
+    uint32_t cr;            /**< (AFEC Offset: 0x00) AFEC Control Register */
+    uint32_t mr;            /**< (AFEC Offset: 0x04) AFEC Mode Register */
+    uint32_t emr;           /**< (AFEC Offset: 0x08) AFEC Extended Mode Register */
+    uint32_t seq1r;         /**< (AFEC Offset: 0x0C) AFEC Channel Sequence 1 Register */
+    uint32_t seq2r;         /**< (AFEC Offset: 0x10) AFEC Channel Sequence 2 Register */
+    uint32_t cher;          /**< (AFEC Offset: 0x14) AFEC Channel Enable Register */
+    uint32_t chdr;          /**< (AFEC Offset: 0x18) AFEC Channel Disable Register */
+    uint32_t chsr;          /**< (AFEC Offset: 0x1C) AFEC Channel Status Register */
+    uint32_t lcdr;          /**< (AFEC Offset: 0x20) AFEC Last Converted Data Register */
+    uint32_t ier;           /**< (AFEC Offset: 0x24) AFEC Interrupt Enable Register */
+    uint32_t idr;           /**< (AFEC Offset: 0x28) AFEC Interrupt Disable Register */
+    uint32_t imr;           /**< (AFEC Offset: 0x2C) AFEC Interrupt Mask Register */
+    uint32_t isr;           /**< (AFEC Offset: 0x30) AFEC Interrupt Status Register */
+    uint32_t _unused1[6];   /** Reserved 0x34 - 0x4B **/
+    uint32_t over;          /**< (AFEC Offset: 0x4C) AFEC Overrun Status Register */
+    uint32_t cwr;           /**< (AFEC Offset: 0x50) AFEC Compare Window Register */
+    uint32_t cgr;           /**< (AFEC Offset: 0x54) AFEC Channel Gain Register */
+    uint32_t _unused2[2];   /** Reserved 0x58 - 0x5F **/
+    uint32_t diffr;         /**< (AFEC Offset: 0x60) AFEC Channel Differential Register */
+    uint32_t cselr;         /**< (AFEC Offset: 0x64) AFEC Channel Selection Register */
+    uint32_t cdr;           /**< (AFEC Offset: 0x68) AFEC Channel Data Register */
+    uint32_t cocr;          /**< (AFEC Offset: 0x6C) AFEC Channel Offset Compensation Register */
+    uint32_t tempmr;        /**< (AFEC Offset: 0x70) AFEC Temperature Sensor Mode Register */
+    uint32_t tempcwr;       /**< (AFEC Offset: 0x74) AFEC Temperature Compare Window Register */
+    uint32_t _unused3[7];   /** Reserved 0x78 - 0x93 **/
+    uint32_t acr;           /**< (AFEC Offset: 0x94) AFEC Analog Control Register */
+} hal_ll_adc_base_handle_t;
+
+/*!< @brief ADC hw specific module values */
+typedef struct
+{
+    uint16_t pin_an[ADC_MODULE_COUNT];
```

Comment:
Use `hal_ll_pin_name_t`. Check other source files.

Reply:
Changed it for all source files where needed.